crates/ruff_python_literal/src/format.rs

```diff
@@ -320,41 +329,61 @@ fn parse_precision(text: &str) -> Result<(Option<usize>, &str), FormatSpecError>
 
 /// Parses a format part within a format spec
 fn parse_nested_placeholder<'a>(
-    parts: &mut Vec<FormatPart>,
+    placeholders: &mut Vec<FormatPart>,
     text: &'a str,
 ) -> Result<&'a str, FormatSpecError> {
     match FormatString::parse_spec(text, AllowPlaceholderNesting::No) {
-        // Not a nested replacement, OK
+        // Not a nested placeholder, OK
         Err(FormatParseError::MissingStartBracket) => Ok(text),
         Err(err) => Err(FormatSpecError::InvalidPlaceholder(err)),
         Ok((format_part, text)) => {
-            parts.push(format_part);
+            placeholders.push(format_part);
             Ok(text)
         }
     }
 }
 
+/// Parse and consume all placeholders in a format spec
+/// This will also consume any intermediate characters such as `x` and `y` in `"x{foo}y{bar}z"`
+/// If no placeholders are present, the text will be returned unmodified.
+fn consume_all_placeholders<'a>(
+    placeholders: &mut Vec<FormatPart>,
+    text: &'a str,
+) -> Result<&'a str, FormatSpecError> {
+    let mut chars = text.chars();
+    let mut text = text;
+    let mut placeholder_count = placeholders.len();
+
+    while chars.clone().contains(&'{') {
+        text = parse_nested_placeholder(placeholders, chars.as_str())?;
+        chars = text.chars();
+        // If we did not parse a placeholder, consume a character
+        if placeholder_count == placeholders.len() {
+            chars.next();
+        } else {
+            placeholder_count = placeholders.len();
+        }
+    }
+    Ok(text)
+}
+
 impl FormatSpec {
     pub fn parse(text: &str) -> Result<Self, FormatSpecError> {
-        let mut replacements = vec![];
-        // get_integer in CPython
-        let text = parse_nested_placeholder(&mut replacements, text)?;
+        let mut placeholders = vec![];
+        let text = consume_all_placeholders(&mut placeholders, text)?;
+
+        if !placeholders.is_empty() {
+            return Ok(FormatSpec::Dynamic(DynamicFormatSpec { placeholders }));
+        }
```

Comment:
Nit: I think you could now consider having `consume_all_placeholders` return a vector rather than taking `&mut Vec` and returning text, since you won't use the remaining text if it returns anything.

Reply:
👍